graphql/query_resolver.go

```diff
@@ -807,9 +808,24 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 	if len(requesters) == 0 {
 		requesters = evergreen.SystemVersionRequesterTypes
 	}
+
+	skipOrderNumber := utility.FromIntPtr(options.SkipOrderNumber)
+	revision := utility.FromStringPtr(options.Revision)
+	if options.SkipOrderNumber == nil && revision != "" {
+		found, err := model.VersionFindOne(model.VersionByProjectIdAndRevisionPrefix(projectId, revision))
+		if err != nil {
+			graphql.AddError(ctx, PartialError.Send(ctx, fmt.Sprintf("could not find git commit '%s'", revision)))
+		} else if found == nil {
+			graphql.AddError(ctx, PartialError.Send(ctx, fmt.Sprintf("could not find git commit '%s'", revision)))
+		} else {
+			// Offset the order number so the specified revision lands nearer to the center of the page.
+			skipOrderNumber = found.RevisionOrderNumber + 1 + limit/2
```

Comment:
This works best if there are mostly uncollapsed commits on the project's waterfall. Then it's likely that this `skipOrderNumber` belongs to an uncollapsed commit, so you will be able to see commits before & after the specified revision. 

However, it's possible for this `skipOrderNumber` to belong to a collapsed commit, in which case you might not see any commits after the specified revision. To get around this I believe I would have to keep "stepping back" and querying until I find an uncollapsed commit, but this is difficult and probably expensive because there could be more than a hundred collapsed commits. To ensure that a commit is not collapsed, I would have to check that that each commit is activated and matches the applied filters (another aggregation pipeline).

At the very least, I think this should guarantee that the commit will be somewhere on the page. Another option would just be to not offset this number, but then you would not ever be able to compare the before & after.